### DISCUSSION TITLE
Easy Way to Auto-Export Every Event/Recording to Exports Folder?

### DISCUSSION THREAD

**Discussion by DC651F82053AD314** [blakeblackshear/frigate#9123] [Ask A Question]
I've upgraded from 0.12-stable to 0.13.0-beta7 and I'm trying to figure how I can auto-export every event/recording to the exports folder right after the recordings finish. I'd like to export all camera recordings in Realtime format. I've used the 'Exports' tab to export manually by the specified date/time, but any pointers as to how to automate the export of _every_ event/recording would be very much appreciated.  Does anyone else out there do this already?

I think the API can do this, but I wondered if there are any examples of what the API calls/requests might look like.  I'm running Frigate in a Docker container -- this means I would run the API calls via the command line within my Frigate Docker container, is that correct?  Would I create a script that runs inside the Docker container to make the 'export all events/recordings' API request?

The idea is, I can then have a script that retains the files in the export folder for X number of days before deleting, and a second script that will watch the export folder and sync all files with my Dropbox as they appear.

Apologies for the noob questions, I'm new to Frigate and new to making API requests within a Docker container in general, but I'm learning lots and really enjoying Frigate!  I was a MotionEye user for years, but Frigate is much better, just takes some time for me to get acclimated to getting it to do what I want. Thanks yall!

**Comment by NickM-27** [comment]
The frigate integration has a service otherwise you'd need something to listen to the MQTT /events and call the API when each event ends

**Reply by DC651F82053AD314** [reply]
When you mention that frigate integration has a service, are you referring to the `GET /api/events` documented here?:
https://docs.frigate.video/integrations/api/#get-apievents

I'd basically want to have every recorded event from all cameras export as soon as each event ends.  I'm currently trying to search for examples of using the API to better understand how I would implement this.

I'm familiar with how to do this via the browser GUI using the new "Exports" tab, but I'd like to automate it.  Any API usage examples/tutorial videos or details you could point me toward would be greatly appreciated!

**Reply by NickM-27** [reply]
> When you mention that frigate integration has a service

no, I am referring to the frigate integration in home assistant having a home assistant service

**Reply by DC651F82053AD314** [reply]
Ah, okay, gotcha.  I'll look into installing Home Assistant for this purpose.  Thanks!

**Reply by DC651F82053AD314** [reply]
Thank you for the continued pointers! Okay, I have installed Home Assistant and I also have the Frigate integration installed within Home assistant.  My Frigate-configured cameras are showing up in Home Assistant, and I see my Frigate recordings showing up in Home Assistant under 'Media' > 'Frigate' > 'Recordings'.

Now, how would I go about exporting all Frigate recordings to a new folder location using Home Assistant?  You mentioned that Home Assistant has a service for this, do you happen to know the name of the service?

If I can get all Frigate recordings exported to a new folder location, I can then use rclone to upload them all to my cloud storage.

**Reply by NickM-27** [reply]
The service is frigate.export I believe

**Reply by DC651F82053AD314** [reply]
Nice, I found the service under the "Developer Tools" section in Home Assistant, it's called "Frigate: Export recording" or `frigate.export_recording`
I'll play around with this and see if I can figure out how to configure it to make a call to export all recordings from both cameras whenever a recording/event has ended.

**Reply by DC651F82053AD314** [reply]
Okay, I'm getting so close to accomplishing my goal now.  I've figured out how to use the `frigate.export_recording` service in Home Assistant to make calls to Frigate for exporting recordings, and it works!  ...

However the recordings are not separated by event like they are in the Frigate interface.  I'd like to export the recordings by event, in the same way they're listed under the "Events" section of the Frigate interface.  The exports I'm able to create are basically all the events from the date range combined into one long video file.

Is there a way to tell the API to separate them all by event?

I used the "UI Mode" to create the calls you see in the YAML in my screenshot below.  What might I add to specify that I'd like all recordings in the date range to be separated by event?

![image](https://github.com/blakeblackshear/frigate/assets/127766978/a9c59225-1aa8-4bb9-bde7-b9de8ff9f149)

**Reply by NickM-27** [reply]
if you want separate exports for each event then just request an export for the time range that each event covers
